pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/HasUserAgedUpRoute.java

```diff
@@ -0,0 +1,73 @@
+package org.broadinstitute.ddp.route;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.ddp.constants.ErrorCodes;
+import org.broadinstitute.ddp.constants.RouteConstants.PathParam;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.db.dao.JdbiUmbrellaStudy;
+import org.broadinstitute.ddp.db.dao.StudyGovernanceDao;
+import org.broadinstitute.ddp.db.dao.UserGovernanceDao;
+import org.broadinstitute.ddp.db.dao.UserProfileDao;
+import org.broadinstitute.ddp.json.errors.ApiError;
+import org.broadinstitute.ddp.model.governance.Governance;
+import org.broadinstitute.ddp.model.governance.GovernancePolicy;
+import org.broadinstitute.ddp.model.user.UserProfile;
+import org.broadinstitute.ddp.pex.PexFetchException;
+import org.broadinstitute.ddp.pex.TreeWalkInterpreter;
+import org.broadinstitute.ddp.util.ResponseUtil;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+/**
+ * Returns the boolean value whether the user has agedUp or not
+ */
+@Slf4j
+@AllArgsConstructor
+public class HasUserAgedUpRoute implements Route {
+
+    @Override
+    public Boolean handle(Request request, Response response) {
+        String userGuid = request.params(PathParam.USER_GUID);
+        String studyGuid = request.params(PathParam.STUDY_GUID);
+
+        log.info("Attempting to retrieve whether the participant:{} agedUp in study:{}",
+                userGuid, studyGuid);
+
+        return TransactionWrapper.withTxn(handle -> {
+            Optional<Governance> governance = handle.attach(UserGovernanceDao.class)
+                    .findGovernancesByParticipantAndStudyGuids(userGuid, studyGuid).findFirst();
+            if (governance.isEmpty()) {
+                return false;
+            }
+            String operatorGuid = governance.get().getProxyUserGuid();
+            Long studyId = handle.attach(JdbiUmbrellaStudy.class)
+                    .getIdByGuid(studyGuid)
+                    .orElseGet(() -> {
+                        String msg = String.format("The study guid '%s' does not refer to a valid study", studyGuid);
+                        log.warn(msg);
+                        throw ResponseUtil.haltError(response, 404, new ApiError(ErrorCodes.STUDY_NOT_FOUND, msg));
+                    });
+
+            GovernancePolicy policy = handle.attach(StudyGovernanceDao.class)
+                    .findPolicyByStudyId(studyId)
+                    .orElse(null);
+            if (policy == null) {
+                throw new PexFetchException(new NoSuchElementException("Governance policy for " + studyGuid + " does not exist"));
```

Comment:
Why is this returning a `PexFetchException`? This is not a PEX error.

Will there be users who are expected to not have a GovernancePolicy, but still have it be within the expected operation of the system? If so, this should _not_ be a 500-class error (as it currently is).

Would a `404` status code with the `ErrorCodes.NOT_FOUND` error and a clear error message for the requestor be applicable here?